tests/unit/s2n_fork_generation_number_test.c

```diff
@@ -319,9 +319,9 @@ static int s2n_test_case_map_inherit_zero_cb(struct fgn_test_case *test_case)
 }
 
 struct fgn_test_case fgn_test_cases[NUMBER_OF_FGN_TEST_CASES] = {
-    { "Default fork detect mechanisms.", s2n_test_case_default_cb, CLONE_TEST_DETERMINE_AT_RUNTIME },
+    { "Default fork detect mechanisms.", s2n_test_case_default_cb_valgrind_marker, CLONE_TEST_DETERMINE_AT_RUNTIME },
     { "Only pthread_atfork fork detection mechanism.", s2n_test_case_pthread_atfork_cb, CLONE_TEST_NO },
-    { "Only madv_wipeonfork fork detection mechanism.", s2n_test_case_madv_wipeonfork_cb, CLONE_TEST_YES },
+    { "Only madv_wipeonfork fork detection mechanism.", s2n_test_case_madv_wipeonfork_cb_valgrind_marker, CLONE_TEST_YES },
```

Comment:
But why would you need to rename the functions? Why not just reference their original names in the suppressions file?

Reply:
> I was not able to resolve the leaks in s2n_fork_generation_number_test and s2n_self_talk_alpn_test and therefore suppressed those warnings.

Since suppression work based on the function name, I wanted to be sure that the name is unique.

Reply:
I prefer to keep the original names. The new names are pretty weird and I don't see any duplicates in the code base currently.

Reply:
> I don't see any duplicates in the code base currently

If duplicates happen, it will be silently suppressed, hence the very unique and odd naming. 

> Are the suppression done based on some regular expression?
String match from what I can tell. `...` matches all lines in between.